tests/displays/cavity_display/frontend/test_decoder.py

```diff
@@ -0,0 +1,246 @@
+import unittest
+from unittest.mock import patch, mock_open
+from PyQt5.QtWidgets import QApplication, QSizePolicy, QAbstractScrollArea
+from PyQt5.QtCore import Qt
+from displays.cavity_display.frontend.decoder import DecoderDisplay, Row, rows
+
+
+
+class TestDecoderDisplay(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Need to create a QApplication instance before running any PyQt tests
+        cls.app = QApplication([])
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.quit()
```

Comment:
Unless this line takes care of it... I'm actually not sure

Reply:
I mean it does clean up the application but there could still be conflicts if multiple test files try to create QApp simultaneously. So definitely safer to just go with your approach I agree. Plus it makes sure only one QApp is ever made.